### PR TITLE
feat: deprecate `Deno.read()` and `Deno.readSync()`

### PR DESCRIPTION
**Pull request by iuioiua**
For removal in Deno v2.

### CODE REVIEW

File: runtime/js/90_deno_ns.js [hunk 1]
```diff
@@ -96,8 +96,22 @@ const denoNs = {
   iter: io.iter,
   iterSync: io.iterSync,
   SeekMode: io.SeekMode,
-  read: io.read,
-  readSync: io.readSync,
+  read(rid, buffer) {
```
Comment (iuioiua) [inline runtime/js/90_deno_ns.js:99]: Doing this hear minimises the impact of the printed alert on unintended APIs.

Comment (bartlomieju) [reply runtime/js/90_deno_ns.js:99]: What do you mean? Are there internal APIs that call this function?

Comment (iuioiua) [reply runtime/js/90_deno_ns.js:99]: Other APIs call the underlying `read()`. E.g. `Deno.stdin.read()` calls that `read()`. In other words, this change purely affects `Deno.read()`. This is the case for other similar APIs.

Not just that, we centralise where all these `warnOnDeprecateApi()` calls are.

**Review by littledivy — APPROVED** [review]
LGTM

**Pull request merged**